modules/services/waybar.nix

```diff
@@ -0,0 +1,164 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.waybar;
+
+  configText = builtins.toJSON ({ inherit (cfg) order; } // cfg.extraConfig);
+
+  configFile = pkgs.writeText "config" configText;
+
+in {
+  options = {
+    services.waybar = {
+      enable = mkEnableOption ''
+        Highly customizable Wayland bar for Sway and Wlroots based compositors
+      '';
+
+      layer = mkOption {
```

Comment:
A number of options here don't seem to actually be used anywhere. To help maintainability I would suggest removing this and all below options and instead have a single `settings` option per [RFC 42](https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#summary).